terraform/monitoring/istio_service_setup.py

```diff
@@ -0,0 +1,60 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import sys
+
+from google.cloud import monitoring_v3
+
+def getIstioServiceName(service_name, project_id, zone):
+	return 'ist:' + project_id + '-zone-' + zone + '-cloud-ops-sandbox-default-' + service_name
+
+def findService(client, service_name, project_id, zone):
+	found_service = False
+	full_service_name = getIstioServiceName(service_name, project_id, zone)
+	service = client.service_path(project_id, full_service_name)
+	while not found_service:
+		try:
+			found_service = client.get_service(service)
+		except:
```

Comment:
Should you be catching a specific exception here?

Reply:
I guess there are 3 different exceptions we want to catch but if there is any kind of error or exception we want to retry the API call anyways - do you think I should specify the exceptions?

Reply:
I think it's fine to be catching generic exception here - given the fact it might run in provisioning. But maybe worth adding a comment saying which exceptions can happen here.

Reply:
Yeah a comment or something might be helpful

Reply:
Added a comment!